src/sage/schemes/elliptic_curves/ell_finite_field.py

```diff
@@ -2414,3 +2416,172 @@ def is_j_supersingular(j, proof=True):
+         Elliptic-curve endomorphism of Elliptic Curve defined by y^2 = x^3 + x over Finite Field in z2 of size 1031^2
+           Via:  (u,r,s,t) = (747*z2 + 284, 0, 0, 0))
+
+        sage: special_supersingular_curve(GF(1033^2), endomorphism=True)
+        (Elliptic Curve defined by y^2 = x^3 + 53*x + 980 over Finite Field in z2 of size 1033^2,
+         Isogeny of degree 11 from Elliptic Curve defined by y^2 = x^3 + 53*x + 980 over Finite Field in z2 of size 1033^2 to Elliptic Curve defined by y^2 = x^3 + 53*x + 980 over Finite Field in z2 of size 1033^2)
+
+        sage: special_supersingular_curve(GF(1039^2), endomorphism=True)
+        (Elliptic Curve defined by y^2 = x^3 + x over Finite Field in z2 of size 1039^2,
+         Elliptic-curve endomorphism of Elliptic Curve defined by y^2 = x^3 + x over Finite Field in z2 of size 1039^2
+           Via:  (u,r,s,t) = (626*z2 + 200, 0, 0, 0))
+
+        sage: special_supersingular_curve(GF(1049^2), endomorphism=True)
+        (Elliptic Curve defined by y^2 = x^3 + 1 over Finite Field in z2 of size 1049^2,
+         Isogeny of degree 3 from Elliptic Curve defined by y^2 = x^3 + 1 over Finite Field in z2 of size 1049^2 to Elliptic Curve defined by y^2 = x^3 + 1 over Finite Field in z2 of size 1049^2)
+
+        sage: special_supersingular_curve(GF(1051^2), endomorphism=True)
+        (Elliptic Curve defined by y^2 = x^3 + x over Finite Field in z2 of size 1051^2,
+         Elliptic-curve endomorphism of Elliptic Curve defined by y^2 = x^3 + x over Finite Field in z2 of size 1051^2
+           Via:  (u,r,s,t) = (922*z2 + 129, 0, 0, 0))
+
+    TESTS::
+
+        sage: p = random_prime(1000)
+        sage: E = special_supersingular_curve(GF(p))
+        sage: E.is_supersingular()
+        True
+        sage: E.order() == p + 1
+        True
+        sage: F.<t> = GF((p,2))
+        sage: E, endo = special_supersingular_curve(F, endomorphism=True)
+        sage: E.is_supersingular()
+        True
+        sage: E.j_invariant() in GF(p)
+        True
+        sage: E.abelian_group().invariants() == (p+1, p+1)
+        True
+        sage: endo.domain() is endo.codomain() is E
+        True
+        sage: endo.trace()
+        0
+        sage: pi = E.frobenius_isogeny()
+        sage: pi.codomain() is pi.domain() is E
+        True
+        sage: pi * endo == -endo * pi
+        True
+
+    .. NOTE::
+
+        This function makes no guarantees about the distribution of
+        the output. The current implementation is deterministic in
+        many cases.
+
+    ALGORITHM: [Bro2009]_, Algorithm 2.4
+    """
+    if not isinstance(F, FiniteField) or (deg := F.degree()) > 2:
+        raise TypeError('input must be a finite field of degree 1 or 2')
+    p = F.characteristic()
+
+    if endomorphism and deg % 2:
+        raise ValueError('endomorphism was requested but is not defined over given field')
+
+    E = None
+
```

Comment:
Can you add a comment to say what q is?  i.e. (I think) it is the degree of the special endo being constructed.

Reply:
It was already mentioned in the docstring, but I also added a comment in the code.